src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/test_vm_commands.py

```diff
@@ -2385,6 +2386,31 @@ def test_create_vm_from_existing_nic(self, resource_group):
         self.assertTrue(re.match(r'[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+', result['publicIps']))
         self.assertTrue(re.match(r'[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+', result['privateIps']))
 
+
+class VMSecretTest(ScenarioTest):
+    @ResourceGroupPreparer()
+    def test_vm_secret_e2e_test(self, resource_group):
+        # a few pre-made key vault artifacts, with the assumption that this test must under team test subscription of 'AzureSDKADGraph2'
+        keyvault = '/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/sdk-test/providers/Microsoft.KeyVault/vaults/vm-secret-kv'
```

Comment:
Why do we need a premade artifact for this?

Reply:
It is about the keyvault DNS issue you mentioned below. On a second thought, I will just abandon the premade, and create the vault on the fly. Extra 60 seconds looks like is needed even i put it before the vm .